grpcgcp/gcp_picker.go

```diff
@@ -101,12 +105,15 @@ func (p *gcpPicker) Pick(info balancer.PickInfo) (balancer.PickResult, error) {
 		}
 		scRef.streamsDecr()
 	}
-	return balancer.PickResult{scRef.subConn, callback}, nil
+	return balancer.PickResult{SubConn: scRef.subConn, Done: callback}, nil
 }
 
 // getSubConnRef returns the subConnRef object that contains the subconn
 // ready to be used by picker.
 func (p *gcpPicker) getSubConnRef(boundKey string) (*subConnRef, error) {
+	p.mu.Lock()
```

Comment:
This is the only used here. Do we need a mutex?

Reply:
I moved it here from the `Pick` which calls `getSubConnRef`. The `Pick` may be called concurrently by different RPC calls.
We need this lock because we are sorting in-place the picker's `scRefs` below (which is, by the way, unnecessary and I'll remove that in the following PRs).